sale_rental/models/sale_order.py

```diff
@@ -142,11 +149,7 @@ def _action_launch_stock_rule(self):
 
                 vals = line._prepare_new_rental_procurement_values(group)
                 try:
-                    self.env['procurement.group'].run(
-                        line.product_id.rented_product_id, line.rental_qty,
-                        line.product_id.rented_product_id.uom_id,
-                        line.order_id.warehouse_id.rental_out_location_id,
-                        line.name, line.order_id.name, vals)
+                    self._run_rental_procurement(line, vals)
```

Comment:
Make sense. :+1: 

Can we call with line instead of self and pass only vals as argument? For example,

    line._run_rental_procurement(vals)

And do necessary changes in generic method.

Reply:
@bodedra Thanks for the review. I have changed the code.